earth_enterprise/rpms/opengee-server/post-install.sh

```diff
@@ -0,0 +1,238 @@
+    configure_publish_root
+
+    # 1) Modify files, maybe should be templated
+    modify_files
+
+    # 2) Set Permissions Before Server Start/Stop
+    fix_postinstall_filepermissions
+
+    # 3) Upgrade Postgres config
+    reset_pgdb    
+
+    # 4) Creating Daemon thread for geserver
+    setup_geserver_daemon
+
+    # 5) Register publish root
+    $BASEINSTALLDIR_OPT/bin/geconfigurepublishroot --noprompt --path=$PUBLISHER_ROOT
+
+    # 6) Install the GEPlaces and SearchExample Databases
+    install_search_databases
+
+    # 7) Repeated step; Set permissions after geserver Start/Stop.
+    fix_postinstall_filepermissions
+
+    # TODO - verify
+    # 8) Run geecheck config script
+    # If file ‘/opt/google/gehttpd/cgi-bin/set_geecheck_config.py’ exists:
+    if [ -f "$GEE_CHECK_CONFIG_SCRIPT" ]; then
+        cd $BASEINSTALLDIR_OPT/gehttpd/cgi-bin
+        python ./set_geecheck_config.py
+    fi
+
+    #9) done!
+    service geserver start
+}
+
+#-----------------------------------------------------------------
+# Post-install Functions
+#-----------------------------------------------------------------
+
+run_as_user() {
+    local use_su=`su $1 -c 'echo -n 1' 2> /dev/null  || echo -n 0`
+    if [ "$use_su" -eq 1 ] ; then
+        >&2 echo "cd / ;su $1 -c \"$2\""
+        ( cd / ;su $1 -c "$2" )
+    else
+        >&2 echo "cd / ;sudo -u $1 $2"
+        ( cd / ;sudo -u $1 $2 )
+    fi
+} 
+
+configure_publish_root()
+{
+    # Update PUBLISHER_ROOT if geserver already installed
+    local STREAM_SPACE="$GEHTTPD_CONF/stream_space"
+    if [ -e $STREAM_SPACE ]; then
+        PUBLISHER_ROOT=`cat $STREAM_SPACE |cut -d" " -f3 |sed 's/.\{13\}$//'`
+    fi
+}
+
+# /etc/init.d/geserver maybe should be expanded from a template instead...
+modify_files()
+{
+    # Replace IA users in a file if they are found.
+    # this step not requied for the OSS installs, however
+    # if there are non-OSS installs in the system, it might help cleanup those.
+
+    # a) Search and replace the below variables in /etc/init.d/geserver.
+    sed -i "s/IA_GEAPACHE_USER/$GEAPACHEUSER/" $BININSTALLROOTDIR/geserver
+    sed -i "s/IA_GEPGUSER/$GEPGUSER/" $BININSTALLROOTDIR/geserver
+
+    # b) Search and replace the file ‘/opt/google/gehttpd/conf/gehttpd.conf’
+    sed -i "s/IA_GEAPACHE_USER/$GEAPACHEUSER/" /opt/google/gehttpd/conf/gehttpd.conf
+    sed -i "s/IA_GEPGUSER/$GEPGUSER/" /opt/google/gehttpd/conf/gehttpd.conf
+    sed -i "s/IA_GEGROUP/$GEGROUP/" /opt/google/gehttpd/conf/gehttpd.conf
+
+    # c) Create a new file ‘/etc/opt/google/fusion_server_version’ and
+    # add the below text to it.
+    echo $RPM_PACKAGE_VERSION >/etc/opt/google/fusion_server_version
+
+    # d) Create a new file ‘/etc/init.d/gevars.sh’ and prepend the below lines.
+    echo -e "GEAPACHEUSER=$GEAPACHEUSER\nGEPGUSER=$GEPGUSER\nGEFUSIONUSER=$GEFUSIONUSER\nGEGROUP=$GEGROUP" >$BININSTALLROOTDIR/gevars.sh
+}
+
+# this too probably should be done inside the rpm...
```

Comment:
The original server install script actually did a lot of this both before and after geserver was started from it, and I am not sure if something in geserver first start made this necessary, so I left most of this in just in case.  Some things like this were left in just to be safe for now.